enterprise/backend/src/metabase_enterprise/serialization/dump.clj

```diff
@@ -24,11 +25,41 @@
 
 (set! *warn-on-reflection* true)
 
+(def ^:private serialization-order
+  (delay (-> (edn/read-string (slurp (io/resource "serialization-order.edn")))
+             (update-keys name)
+             (update-vals (fn [order]
+                            (into {} (map vector order (range))))))))
+
+(def ^:private serialization-sorted-map
+  (memoize
+   (fn [t]
+     (if-let [order (get @serialization-order t)]
+       ;; known columns are sorted by their order, then unknown are sorted alphabetically
+       (let [getter #(if (contains? order %)
+                       [0 (get order %)]
+                       [1 %])]
+         (sorted-map-by (fn [k1 k2]
+                          (compare (getter k1) (getter k2)))))
+       (sorted-map)))))
+
+(defn- serialization-deep-sort [m]
+  (let [t    (-> (:serdes/meta m) last :model)
+        smap (serialization-sorted-map t)]
+    (into smap
+          (update-vals m (fn [val]
+                           (cond
+                             (map? val)               (serialization-deep-sort val)
```

Comment:
Embedded things like dashcards generally don't have `:serdes/meta` on them, so they won't be ordered with a meaningful order, only alphabetically. That's not so bad, but I think we can do better.

Probably the best way to make this work is to change the `model-name -> ordering` mapping above to `[model-name :key :path] -> ordering`. Then `["Dashboard"]` gives the ordering for a Dashboard from the file, and `["Dashboard" :ordered_cards]` can return the ordering for `DashboardCard`s (which isn't in the file yet but should be). Then this recursion would keep track of the path as it goes. (Note that we don't care about indexes in sequences, just about nesting in maps.)

Reply:
Done in the last commit, works for both the path (like `[:Dashboard :ordered_cards]`) or for the last key in the path (`:serdes/meta`).